source/ShipInfoDisplay.cpp

```diff
@@ -240,31 +240,18 @@ void ShipInfoDisplay::UpdateAttributes(const Ship &ship, const Depreciation &dep
 		attributesHeight += 20;
 	}
 	
-	if(ship.HasBays())
-	{
-		unsigned drone = 0;
-		unsigned fighter = 0;
-		for(const auto &bay : ship.Bays())
-		{
-			if(bay.isFighter)
-				++fighter;
-			else
-				++drone;
-		}
-		
-		if(drone)
+	// Print the number of bays for each bay-type we have
+	for(auto bayType : Ship::BAY_TYPES)
+		if(ship.BaysFree(bayType))
```

Comment:
Use of `BaysFree` won't properly display bay counts while in flight

Reply:
Agreed, using the new BaysTotal function.